packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3159,6 +3165,9 @@ export class ContainerRuntime
 				this.pendingStateManager.popStagedBatches(({ content, localOpMetadata }) =>
 					this.rollback(content, localOpMetadata),
 				);
+				if (this.attachState === AttachState.Attached) {
+					this.updateDocumentDirtyState(this.pendingMessagesCount !== 0);
```

Comment:
Yeah looking at callers to this, I think we could be more rigid and just tie it to PSM state.